### PR TITLE
doc(report_quiz): added a `README.md`

### PR DESCRIPTION
**Pull request by HanslettTheDev**
Documentation for the report quiz bot

### CODE REVIEW

**Comment by pythonbrad** [comment]
@HanslettTheDev 
Thanks for this documentation.

I will leave some reviews as possible

File: .github/workflows/report_quiz/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+## REPORT QUIZ BOT
+
+[![Build Status](https://github.com/osscameroon/report-quiz/actions/workflows/main.yml/badge.svg)](https://github.com/osscameroon/report-quiz/actions/workflows/main.yml)
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: Is this, the link to the osscameroon report_quiz GitHub action status?

you can use it

[telegram github quiz](https://github.com/osscameroon/global-github-actions/actions/workflows/report_a_quiz.yaml)

```
[![telegram github quiz](https://github.com/osscameroon/global-github-actions/actions/workflows/report_a_quiz.yaml/badge.svg)](https://github.com/osscameroon/global-github-actions/actions/workflows/report_a_quiz.yaml)
```

File: .github/workflows/report_quiz/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+## Description
+
+This is an automated workflow for the report quiz bot. This quiz bot performs the following actions
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: All the repo is about workflows.
What do you think about "this workflow is to automate the management of quiz competitions"?

File: .github/workflows/report_quiz/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+- Selects a random quiz from https://quizapi.io/api/ website
+- It periodically sends a message in the group to start a quiz competition
+- It stores the results of each quiz inside of a json file located **archive** folder
+- Once the quiz is over, it sends a message in the group with the results
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: A description with lists. 🤔 
It looks imperative.
I think a paragraph is better.

File: .github/workflows/report_quiz/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+### How to run this locally
+
+1. Clone the repository
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md:20]: The git command is missing

File: .github/workflows/report_quiz/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+2. Locate the .github/workflows/report-quiz folder in the root of the repository
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: The cd command is missing

File: .github/workflows/report_quiz/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+4. Edit the env.sh file with the necessary data
+
+   - Edit the **TELEGRAM_BOT_TOKEN** variable
+      - **How to get a telegram BOT Token**
+         - Visit https://telegram.me/BotFather
+         - Use the command **/newbot** and follow the instructions to create a new bot
+         - Copy the token from @BotFather telegram bot and paste the token in the env.sh file
+
+   - Edit the **TELEGRAM_GROUP_ID** variable
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: The tutorial of telegram is well explained.
I don't believe that we need to explain it here

File: .github/workflows/report_quiz/README.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        - You remember the bot you just created? Now take that bot and add it into any group of your choice (How to add a bot to a telegram group)[https://stackoverflow.com/questions/37338101/how-to-add-a-bot-to-a-telegram-group]
+        - Now visit this site `https://api.telegram.org/bot[BOT_TOKEN_HERE]/getMe`
+        - replace `[BOT_TOKEN_HERE]` with the bot token you got from bot BotFather
+        - When the page opens ensure you can see a page looking like this
+        - ![image](https://github.com/user-attachments/assets/9dadba18-154a-479c-a870-adaf63469fa4)
+        - Now go to the group you just added the bot and use this command `/start@THE_USERNAME_OF_YOUR_BOT`
+        - Replace `THE_USERNAME_OF_YOUR_BOT` with your bots username
+        - Now open this link `https://api.telegram.org/bot[BOT_TOKEN_HERE]/getUpdates` and you should see a json object called results. Look for any key with the name **chat** and under it's child elements you will see **id**. Copy the **id** including the `-`(minus) sign in front of the id
+        - ![image](https://github.com/user-attachments/assets/610217af-2ed8-4309-a59d-2eea5399ed99)
+
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: I suppose a link is enough.

File: .github/workflows/report_quiz/README.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+   - Edit the **QUIZAPI_KEY** variable
+      - **How to get a quizapi key**
+        - Visit https://quizapi.io/
+        - Create an account
+        - Visit https://quizapi.io/api/ page and generate a new key. Copy the key and paste it in the env.sh file
+
+5. Setting up your bash terminal to run the bot
+
+   - Run the following command in your terminal. Ensure that you have `jq` and `git` installed, if not install them using `sudo apt install jq git`
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: This part, can be a section called dependencies

File: .github/workflows/report_quiz/README.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+
+   ```bash
+   source main.sh
+   chmod +x main.sh
+   ```
+
+
+   - Now to run this bot, you need to run the commands in a following order for them to work normally. All the functions are located in `.github/workflows/report_quiz/quiz/functions.sh` file
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: `quiz/functions.sh` are functions related only to the quiz (normal quiz).
They are others `functions.sh` file in the code.

File: .github/workflows/report_quiz/README.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+
+   ```bash
+   . main.sh; start_quiz_competition # Use this command to send/initiate a quiz competition in the group
+   . main.sh; fetch_quiz_users_answers # Use this command to fetch the users answers before stoping the quiz
+   . main.sh; stop_quiz_competition # Use this command to stop the quiz and to prepare for sending results
```
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md]: It's an important point, what about add a section for it (e.g "main command")

File: .github/workflows/report_quiz/README.md
Comment (pythonbrad) [inline .github/workflows/report_quiz/README.md:1]: I suggests this structure

- description
- requirements
- how to setup
- how to run
- useful resources

**Comment by HanslettTheDev** [comment]
I took a look at it and I don't think this type of project requires that many sections

The instructions are detailed in point format, which should be clear to a user that you follow the instructions from number 1 to 5

You mentioned about highlighting relevant commands, I checked, and it seems I highlighted the important commands. I even mentioned about the aspect that they need to run the commands in a specific order

**Review by pythonbrad — APPROVED** [review]
With your permission, I applied the changes.

**Pull request merged**